src/main/kotlin/org/rust/ide/annotator/fixes/NameSuggestionFix.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.annotator.fixes
+
+import com.intellij.codeInspection.LocalQuickFixOnPsiElement
+import com.intellij.grazie.speller.utils.Distances.levenshtein
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import org.rust.lang.core.psi.RsPath
+
+/**
+ * Changes the text of some element to the suggested name using the provided function.
+ */
+class NameSuggestionFix(
+    element: PsiElement,
+    private val newName: String,
+    private val replaceElementFn: (name: String) -> PsiElement?
+): LocalQuickFixOnPsiElement(element) {
+    override fun getFamilyName(): String = "Change name of element"
+    override fun getText(): String = "Change to `$newName`"
+
+    override fun invoke(project: Project, file: PsiFile, startElement: PsiElement, endElement: PsiElement) {
+        val newElement = replaceElementFn(newName) ?: return
+        startElement.replace(newElement)
+    }
+
+    companion object {
+        fun createApplicable(
+            element: RsPath,
+            name: String,
+            validNames: List<String>,
+            replaceElementFn: (name: String) -> PsiElement?,
+            maxDistance: Int = 1
```

Comment:
If you swap `replaceElementFn` and `maxDistance` with each other, you can pass lambda directly outside of the parenthesis on call site. I think it's a little bit more idiomatic